app/models/wallet.py

```diff
@@ -41,3 +43,56 @@ class IndyCredInfo(IndyCredInfoAcaPy):
 
 class CredInfoList(BaseModel):
     results: Optional[List[IndyCredInfo]] = None
+
+
+class DIDCreate(DIDCreateAcaPy):
+    """
+    Extends the AcapyDIDCreate model with smart defaults and simplified interface.
+    Automatically handles the options field structure while maintaining compatibility.
+    """
+
+    method: Optional[StrictStr] = Field(
+        default="sov",
+        description="Method for the requested DID. Supported methods are 'sov', `web`, `did:peer:2` or `did:peer:4`.",
+        examples=["sov", "key", "did:peer:2", "did:peer:4"],
+    )
+    options: Optional[DIDCreateOptions] = Field(
+        default=None,
+        deprecated=True,
+        description="To define a key type and/or a did depending on chosen DID method.",
+        examples=[{"key_type": "ed25519", "did": "did:peer:2"}],
+    )
+    seed: Optional[StrictStr] = Field(
+        default=None,
+        description="Optional seed to use for DID. Must be enabled in configuration before use.",
+    )
+    key_type: Optional[StrictStr] = Field(
+        default="ed25519",
+        description="Key type to use for the DID key_pair. Validated with the chosen DID method's supported key types.",
+        examples=["ed25519", "bls12381g2"],
+    )
+    did: Optional[str] = Field(
+        default=None,
+        description="Specify final value of did (including did:<method>: prefix) if the method supports/requires it.",
+        strict=True,
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def handle_deprecated_options(cls, values: dict) -> dict:
+        """
+        Handle both deprecated options field and new flattened fields.
+        Priority: If both are provided, new fields take precedence.
+        """
+
+        if values.get("options") is None:
+            values["options"] = {}
+            values["options"]["key_type"] = values.get("key_type") or "ed25519"
+            values["options"]["did"] = values.get("did")
+        else:
+            if not values["options"]["key_type"]:
+                values["options"]["key_type"] = values.get("key_type") or "ed25519"
+            if not values["options"].get("did") and values.get("did"):
+                values["options"]["did"] = values.get("did")
```

Comment:
If we're in the else block, then options is None, so the secondary if checks are redundant. Funny enough, they're not safe! Since the top if calls .get("options") and here the keys are just being accessed, so it could raise a KeyError.

I think we can just drop the secondary `if not` statements.

And `values.get("key_type") or "ed25519"` can just be `values.get("key_type", "ed25519")`

Reply:
Hey I updated the logic here let me know what you think